### PR TITLE
add schema support

### PR DESCRIPTION
**Pull request by eugenemiretsky**

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf [hunk 1]
```diff
@@ -94,6 +94,13 @@ resource "google_pubsub_topic" "topic" {
       allowed_persistence_regions = message_storage_policy.key == "allowed_persistence_regions" ? message_storage_policy.value : null
     }
   }
+  dynamic "schema_settings" {
+    for_each = var.schema_settings
```
Comment (morgante) [inline main.tf:98]: This is going to loop over the values in `schema_settings` - probably not what you expected.

Please add this to an example and test to make sure your logic is correct.

**Review by morgante — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (eugenemiretsky) [reply main.tf]: Hey! Change schema_settings to be an object

Comment (eugenemiretsky) [reply main.tf]: Hey @morgante,  any feedback on the new changes?

Comment (eugenemiretsky) [reply main.tf]: @morgante Can we please review an merge this?

File: test/integration/pubsub/controls/pubsub.rb [hunk 2]
```diff
@@ -37,3 +39,9 @@
   it { expect(stdout).to include(pushConfig: { pushEndpoint: "https://#{project_id}.appspot.com/" }) }
   it { expect(stdout).to include(ackDeadlineSeconds: 20) }
 end
+
+describe command("gcloud --project='#{project_id}' pubsub schemas describe #{schema_name}") do
```
Comment (morgante) [inline test/integration/pubsub/controls/pubsub.rb:43]: This test is failing.

```
>>>>>>     Failed to complete #verify action: [/workspace/test/integration/pubsub/controls/pubsub.rb:44: syntax error, unexpected tIDENTIFIER, expecting '('
...it_status) { should not be_zero }
...                        ^~~~~~~] on pubsub-local
```

File: examples/simple/main.tf [hunk 3]
```diff
@@ -18,6 +18,13 @@ provider "google" {
   region = "us-central1"
 }
 
+resource "google_pubsub_schema" "example" {
```
Comment (morgante) [inline examples/simple/main.tf:21]: Should we actually just move this into the module? So an input variable could be:

```
schema = {
  name = "example"
  type = "AVRO"
  definition = "{\n \"type\" : \"record\",\n \"name\" : \"Avro\",\n \"fields\" : [\n {\n \"name\" : \"StringField\",\n \"type\" : \"string\"\n },\n {\n \"name\" : \"IntField\",\n \"type\" : \"int\"\n }\n ]\n}\n"
  encoding = "JSON"
}
```

Just a thought.

**Review by morgante — CHANGES_REQUESTED** [review]

**Comment by akovtun-extenda** [comment]
Hi! Any updates on this one? We'd like to have schemas support to be merged as well.

**Comment by eugenemiretsky** [comment]
@morgante mind taking another look?

**Review by morgante — APPROVED** [review]

**Pull request merged**